### PR TITLE
refactor HazelCast

### PR DESCRIPTION
**Pull request by carolinaloureiro**
Add AllowParallelizationSettings Trait and add new methods to KcqlSettings.

### CODE REVIEW

File: kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkConfig.scala [hunk 1]
```diff
@@ -69,9 +69,11 @@ class HazelCastSinkConfig(props: util.Map[String, String])
     with ErrorPolicySettings
     with KcqlSettings
     with NumberRetriesSettings
-with ThreadPoolSettings{
+    with ThreadPoolSettings
+with AllowParallelizationSettings{
```
Comment (fartariatomas) [inline kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkConfig.scala]: should be inline with previous lines

File: kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkSettings.scala [hunk 2]
```diff
@@ -54,39 +53,40 @@ object HazelCastSinkSettings {
...
     val threadPoolSize = config.getThreadPoolSize
+    val topicTables = getTopicTables(routes)
+
+    ensureGroupNameExists(config)
 
+    val connConfig = HazelCastConnectionConfig(config)
+    val client = HazelCastConnection.buildClient(connConfig)
 
-    //    routes.map(r => (r.getSource, r.getTarget)).toMap
+    new HazelCastSinkSettings(client, routes, topicTables, fieldMap, ignoreFields, primaryKeys, errorPolicy, maxRetries, format, threadPoolSize, allowParallel)
```
Comment (fartariatomas) [inline kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkSettings.scala]: line too long

**Review by joseluisvf — COMMENTED** [review]
Could not have done it any better

**Review by joseluisvf — APPROVED** [review]

File: kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkConfig.scala [hunk 3]
```diff
@@ -18,7 +18,7 @@ package com.datamountaineer.streamreactor.connect.hazelcast.config
 
 import java.util
 
-import com.datamountaineer.streamreactor.temp.{ErrorPolicySettings, KcqlSettings, NumberRetriesSettings, ThreadPoolSettings}
+import com.datamountaineer.streamreactor.temp._
```
Comment (DavidNavalho) [inline kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkConfig.scala]: It's generally a bad practice for 'wildcard' imports. Are you using more then those 4 imports?

**Review by fartariatomas — APPROVED** [review]
Looks good

File: kafka-connect-rethink/src/main/scala/com/datamountaineer/streamreactor/connect/rethink/config/ReThinkSinkConfig.scala [hunk 4]
```diff
@@ -18,7 +18,7 @@ package com.datamountaineer.streamreactor.connect.rethink.config
 
 import java.util
 
-import com.datamountaineer.streamreactor.temp.{ErrorPolicySettings, KcqlSettings, NumberRetriesSettings}
+import com.datamountaineer.streamreactor.temp._
```
Comment (DavidNavalho) [inline kafka-connect-rethink/src/main/scala/com/datamountaineer/streamreactor/connect/rethink/config/ReThinkSinkConfig.scala:21]: Same as before

File: kafka-connect-temp/src/main/scala/com/datamountaineer/streamreactor/temp/KcqlSettings.scala [hunk 5]
```diff
@@ -21,4 +21,16 @@ trait KcqlSettings extends BaseSettings {
   def getIgnoreFields(routes: Set[Config] = getRoutes): Map[String, Set[String]] = {
     routes.map(rm => (rm.getSource, rm.getIgnoredField.toSet)).toMap
   }
+
+  def getPrimaryKeys(routes: Set[Config] = getRoutes): Map[String, Set[String]] = {
+    routes.map(r => (r.getSource, r.getPrimaryKeys.toSet)).toMap
```
Comment (DavidNavalho) [inline kafka-connect-temp/src/main/scala/com/datamountaineer/streamreactor/temp/KcqlSettings.scala]: If I may be picky, why the use of 'r' variable in some places, and 'rm' in another? 
- no need to actually extend this variable name for this scenario, just something in terms of consistency - pick one? =)

**Review by DavidNavalho — COMMENTED** [review]
Looks good - only some style comments

File: kafka-connect-temp/src/main/scala/com/datamountaineer/streamreactor/temp/KcqlSettings.scala (hunk 5 above)
Comment (carolinaloureiro) [reply kafka-connect-temp/src/main/scala/com/datamountaineer/streamreactor/temp/KcqlSettings.scala]: thanks, done!

File: kafka-connect-rethink/src/main/scala/com/datamountaineer/streamreactor/connect/rethink/config/ReThinkSinkConfig.scala (hunk 4 above)
Comment (carolinaloureiro) [reply kafka-connect-rethink/src/main/scala/com/datamountaineer/streamreactor/connect/rethink/config/ReThinkSinkConfig.scala:21]: can't change this given it's not hazelcast related but i'll spread the message.

File: kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkConfig.scala (hunk 3 above)
Comment (carolinaloureiro) [reply kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/config/HazelCastSinkConfig.scala]: yes, IntelliJ does this if line gets too long. I've separated it into newlines

File: kafka-connect-rethink/src/main/scala/com/datamountaineer/streamreactor/connect/rethink/config/ReThinkSinkConfig.scala (hunk 4 above)
Comment (DavidNavalho) [reply kafka-connect-rethink/src/main/scala/com/datamountaineer/streamreactor/connect/rethink/config/ReThinkSinkConfig.scala:21]: It's not something I'm against - if you're using implicit conversions, or most of the imports, then a wildcard import is more then OK - they shouldn't have an impact on performance...if it's only a couple (2,3...4...), then I typically prefer them separate, like they were before ({ErrorPolicySettings, KcqlSettings, NumberRetriesSettings})

**Review by pedrobento988 — APPROVED** [review]

**Pull request merged**